src/compiler/checker.ts

```diff
@@ -17430,6 +17436,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             for (const t of u.types) {
                 if (insertType(checked, t)) {
                     if (eachUnionContains(unionTypes, t)) {
+                        // undefinedType/missingType should always come sorted first so we leverage that here
+                        if (t === undefinedType && result.length && result[0] === missingType) {
+                            continue;
+                        }
+                        if (t === missingType && result.length && result[0] === undefinedType) {
+                            result[0] = missingType;
+                            continue;
+                        }

```

Comment:
I used this PR as test for some new code coverage stuff, and it actually revealed that this code is never tested: https://app.codecov.io/gh/jakebailey/TypeScript-codecov-testing/pull/3

Reply:
I pushed out a new test case and verified manually that those lines are hit now